packages/hub/src/lib/cache-management.ts

```diff
@@ -0,0 +1,252 @@
+import { homedir } from "node:os";
+import { join, basename } from "node:path";
+import { stat, readdir, readFile, realpath, lstat } from "node:fs/promises";
+import type { Stats } from "node:fs";
+
+const default_home = join(homedir(), ".cache");
```

Comment:
on `deno` this causes issues because it does a sys call (see e2e tests)

it should only be called as needed, not top-level, so that a program that does not need this functionality does not need the extra deno permission